src/gmail.js

```diff
@@ -1166,7 +1167,12 @@ var Gmail = function(localJQuery) {
         if (threadData && api.check.data.is_action(threadData)) {
             const action = api.tools.get_action(threadData);
 
-            return action_map[action];
+            //Check if label is applied to email / existing email is moved to an label
+            if(action.startsWith(label) && api.check.data.is_first_type_action(threadData)) {
+                return action_map[label];
+            } else {
+                return action_map[action];
+            }
         } else {
```

Comment:
Are these changes related to the feature being implemented? If so, it's not obvious how.

Could you clarify?